frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3569,9 +3569,30 @@ static bool resolveFnCallSpecial(Context* context,
                                  const AstNode* astForErr,
                                  const CallInfo& ci,
                                  QualifiedType& exprTypeOut) {
-  // TODO: .borrow()
   // TODO: chpl__coerceCopy
 
+  // Consider the resolution of .borrow() and release here a performance optimization, other
+  // methods on _owned should be resolved the "normal" way, through the module code
+  if (ci.isMethodCall() &&
+      (ci.name() == UniqueString::get(context, "borrow") ||
+       ci.name() == UniqueString::get(context, "release"))) {
+    if (ci.numActuals() == 2) {
+      auto thisQt = ci.actual(0).type();
+      auto actualQt = ci.actual(1).type();
+      if (thisQt.type()->isAnyOwnedType() && actualQt.type()->isClassType()) {
+        if (actualQt.type()->toClassType()->decorator().isManaged()) {
+          // TODO: better checking that this is our OwnedObject.borrow/release method?
+          if (auto manager = actualQt.type()->toClassType()->manager()) {
+            if (manager->isAnyOwnedType()) {
+              exprTypeOut = actualQt;
+              return true;
```

Comment:
Am I reading this right that `owned.borrow(x)` will produce an `owned C` if `x` was owned? Doesn't seem right. I also don't see any mention of a type-based `borrow` method like the one you're implementing here...

That is, I don't see a `borrow` method with two actuals at all.

Reply:
Also, `.release` should return `unmanaged`, no?

Reply:
I think you need tests of this logic as part of this PR to make the intent clear.